src/login/logind-action.c

```diff
@@ -133,6 +133,44 @@ const HandleActionData* handle_action_lookup(HandleAction action) {
         return &handle_action_data_table[action];
 }
 
+static const HandleAction sleep_actions[] = {
+        HANDLE_SUSPEND_THEN_HIBERNATE,
+        HANDLE_HYBRID_SLEEP,
+        HANDLE_HIBERNATE,
+        HANDLE_SUSPEND,
```

Comment:
hmm, so i think the delay that hibernates causes means HANDLE_SUSPEND should come 2nd, not last.

Reply:
Hmm, my idea was that we should always prefer "advanced" or "safe" sleep operations over simple ones, because `SLEEP_SUSPEND` is almost always supported. If we order it before the two hibernation operations, especially `hybrid-sleep`, then they might never be used at all...

I'm fine with removing `hibernate` from the default though, i.e. only use `s2h` and `suspend` by default.

Reply:
I think it's fine to never use hybrid-sleep.

The latency when going to sleep and when coming back that hibernation is I think is prohibitve enough to not use it by default. People are used to the fact that suspend/resume is quick, and if we by default start doing hybrid sleep and thus put the latency on everyone (just becuase s2h is not available) this sounds highly problematic to me. I think it's only a safe choice if there's nothing else, or if it is used as "backup" like in s2h.

Hence i'd put s2h first, followed by suspend, followed by hibernate. and not pick hybrid-sleep at all

Reply:
Oh i get it now. you hardcode the order, but let the user pick the mask. that selects the subset of it. i think the approach is ok, but i'd still swap hibernate and suspend.

hibernate usually requires less hw support, and comes with the latency issue. I'd really but that last.

hence i am for this order:

s2h → hybrid-sleep → suspend → hibernate

and the default mask should be s2h|suspend|hibernate